### PR TITLE
treewide: move astro libraries into pkgs/development/libraries/science/astronomy

### PR DESCRIPTION
**Pull request by hjones2199**
###### Motivation for this change
Science libraries are mostly grouped in nixpkgs in the pkgs/development/libraries/science/<field_name> directory. This PR brings astronomy libraries into the fold, wrangling them under pkgs/development/libraries/science/astronomy.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hjones2199** [comment]
Not sure of the way to name the commit when multiple packages are being moved, or if moving existing packages to a different directory is acceptable in nixpkgs. That said, it seemed like a good idea to propose that astro libraries follow the same pattern as other science fields by going in pkgs/development/libraries/astronomy.

**Comment by SuperSandro2000** [comment]
I would name it ``treewide: move astro libraries into X``

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -17618,7 +17618,7 @@ in
 
   waylandpp = callPackage ../development/libraries/waylandpp { };
 
-  wcslib = callPackage ../development/libraries/wcslib { };
+  wcslib = callPackage ../development/libraries/science/astronomy/wcslib { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:17621]: This package has no maintainer set.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -14782,9 +14782,9 @@ in
   indicator-application-gtk2 = callPackage ../development/libraries/indicator-application/gtk2.nix { };
   indicator-application-gtk3 = callPackage ../development/libraries/indicator-application/gtk3.nix { };
 
-  indilib = callPackage ../development/libraries/indilib { };
-  indi-3rdparty = callPackage ../development/libraries/indilib/indi-3rdparty.nix { };
-  indi-full = callPackage ../development/libraries/indilib/indi-full.nix { };
+  indilib = callPackage ../development/libraries/science/astronomy/indilib { };
+  indi-3rdparty = callPackage ../development/libraries/science/astronomy/indilib/indi-3rdparty.nix { };
+  indi-full = callPackage ../development/libraries/science/astronomy/indilib/indi-full.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:14787]: This package has no maintainer set.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (hjones2199) [reply pkgs/top-level/all-packages.nix:17621]: now adopted in subsequent commit

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (hjones2199) [reply pkgs/top-level/all-packages.nix:14787]: Thanks, did some digging on how to fix that in a buildEnv derivation, fixed in latest commit by inheriting meta from base indilib.

**Comment by hjones2199** [comment]
Result of `nixpkgs-review pr 118111` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>kstars</li>
    <li>siril</li>
    <li>stellarsolver</li>
    <li>wcslib</li>
  </ul>
</details>

**Pull request merged**